frontend/app-development/layout/SettingsModalButton/SettingsModal/SettingsModal.tsx

```diff
@@ -69,30 +53,40 @@ export type SettingsModalProps = {
  *
  * @property {boolean}[isOpen] - Flag for if the modal is open
  * @property {function}[onClose] - Function to be executed on close
- * @property {Policy}[policy] - The policy of the app
  * @property {string}[org] - The org
  * @property {string}[app] - The app
- * @property {AppConfig}[appConfig] - The serice name
- * @property {Repository}[repository] - The repository of the app
- * @property {strign}[createdBy] - The name of the user that created the app
- * @property {AppConfig}[appConfig] - The service name
- * @property {ApplicationMetadata}[appMetadata] - The application's metadata
  *
  * @returns {ReactNode} - The rendered component
  */
-export const SettingsModal = ({
-  isOpen,
-  onClose,
-  policy,
-  org,
-  app,
-  appConfig,
-  repository,
-  createdBy,
-  appMetadata,
-}: SettingsModalProps): ReactNode => {
+export const SettingsModal = ({ isOpen, onClose, org, app }: SettingsModalProps): ReactNode => {
   const { t } = useTranslation();
 
+  const {
+    status: policyStatus,
+    data: policyData,
+    error: policyError,
+  } = useAppPolicyQuery(org, app);
+  const {
+    status: appConfigStatus,
+    data: appConfigData,
+    error: appConfigError,
+  } = useAppConfigQuery(org, app);
+  const {
+    status: repositoryStatus,
+    data: repositoryData,
+    error: repositoryError,
+  } = useRepoMetadataQuery(org, app);
+  const {
+    status: initialCommitStatus,
+    data: initialCommitData,
+    error: initialCommitError,
+  } = useRepoInitialCommitQuery(org, app);
+  const {
+    status: appMetadataStatus,
+    data: appMetadataData,
+    error: appMetadataError,
+  } = useAppMetadataQuery(org, app);
```

Comment:
Her går det potensielt mange spørringer på en gang. Det er ikke nødvendigvis feil, men har du vurdert om det er mulig å flytte noen av dem til en underkomponent?

Reply:
Det er god praksis å vente med hente data til brukeren spør om det. Så det å flytte kall som kan flyttes ned er lurt, tenker jeg. 😊 

Det er også fint å ha i bakhode at en nettleser gjør kun 6 kall parallelt og resten blir satt i en kø, som gjør at mange kall samtidig kan oppleves som tregt for brukeren.